src/v/bytes/BUILD

```diff
@@ -0,0 +1,132 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+
+redpanda_cc_library(
+    name = "iobuf",
+    srcs = [
+        "iobuf.cc",
+    ],
+    hdrs = [
+        "include/bytes/details/io_allocation_size.h",
+        "include/bytes/details/io_byte_iterator.h",
+        "include/bytes/details/io_fragment.h",
+        "include/bytes/details/io_iterator_consumer.h",
+        "include/bytes/details/io_placeholder.h",
+        "include/bytes/details/out_of_range.h",
+        "include/bytes/iobuf.h",
+    ],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"],
```

Comment:
FWIW instead of making everything public, you can also make public the default.

```starlark
package(
  default_visibility = ["//visibility:public"],
)
```

https://bazel.build/reference/be/functions#package

Also at somepoint we probably want to limit visibility, but it's probably fine for now.

Reply:
ahh right. honestly after this PR merges I think it's gonna be open season on evolving the Bazel build. Hopefully we put down a solid foundation!

Reply:
I think this is a good foundation - I am happy to merge and help tweak stuff as we go along.